manifests/server/role.pp

```diff
@@ -47,134 +48,181 @@
     } else {
       $role_connection_limit = $connection_limit
     }
-    $password_table = "pg_user"
+    $password_table = "pg_shadow"
 
   } else {
     fail("dialect must be set to a valid value")
   }
 
-  # If possible use the version of the remote database, otherwise
-  # fallback to our local DB version
-  if $connect_settings != undef and has_key( $connect_settings, 'DBVERSION') {
-    $version = $connect_settings['DBVERSION']
-  } else {
-    $version = $postgresql::server::_version
-  }
-
-  $login_sql       = $login       ? { true => 'LOGIN',       default => 'NOLOGIN' }
-  $inherit_sql     = $inherit     ? { true => 'INHERIT',     default => 'NOINHERIT' }
-  $createrole_sql  = $createrole  ? { true => "CREATE${role_keyword}",  default => "NOCREATE${role_keyword}" }
-  $createdb_sql    = $createdb    ? { true => 'CREATEDB',    default => 'NOCREATEDB' }
-  $superuser_sql   = $superuser   ? { true => 'SUPERUSER',   default => 'NOSUPERUSER' }
-  $replication_sql = $replication ? { true => 'REPLICATION', default => '' }
-  if ($password_hash != false) {
-    $environment  = "NEWPGPASSWD=${password_hash}"
-    if ($dialect == 'postgres') {
-      $password_sql = "ENCRYPTED PASSWORD '\$NEWPGPASSWD'"
-    } elsif ($dialect == 'redshift') {
-      # Redshift does not define the ENCRYPTED keyword to interpret a hash
-      $password_sql = "PASSWORD '\$NEWPGPASSWD'"
+  if $ensure == 'absent' {
+    Postgresql_psql {
+      db         => $db,
+      port       => $port_override,
+      psql_user  => $psql_user,
+      psql_group => $psql_group,
+      psql_path  => $psql_path,
+      connect_settings => $connect_settings,
+      cwd        => $module_workdir,
+      require    => [
+        Postgresql_psql["${title}: DROP ${role_keyword} ${username}"],
+        Class['postgresql::server'],
+      ],
+    }
+  
+    postgresql_psql { "${title}: DROP ${role_keyword} ${username}":
+      command     => "DROP ${role_keyword} \"${username}\"",
+      unless      => "SELECT 1 WHERE NOT EXISTS (SELECT 1 FROM ${role_table} WHERE ${role_column_prefix}name = '${username}')",
+      environment => $environment,
+      require     => Class['Postgresql::Server'],
     }
   } else {
+    # If possible use the version of the remote database, otherwise
+    # fallback to our local DB version
+    if $connect_settings != undef and has_key( $connect_settings, 'DBVERSION') {
+      $version = $connect_settings['DBVERSION']
+    } else {
+      $version = $postgresql::server::_version
+    }
+  
     if ($dialect == 'postgres') {
-      $password_sql = ''
+      $login_sql       = $login       ? { true => 'LOGIN',       default => 'NOLOGIN' }
     } elsif ($dialect == 'redshift') {
-      $password_sql = "PASSWORD DISABLE"
+      if ($login != false) {
+        $login_sql = $login
```

Comment:
I'm a little confused about this, since `$login` is a bool, if I call the module with `$login` = true, you're going to set the `$login_sql` to the boolean `true`?

Wouldn't it be better to do something similar to what you're doing for the postgres dialect above?